docs/guide/docs/infrastructure/development-pipeline.md

```diff
@@ -0,0 +1,115 @@
+---
+id: development-lifecycle
+title: Development Lifecycle
+---
+
+## Git Syncing – Source Control Management
+
+A best practice is to keep track of changes to your chatbot using your preferred [Source Control Management Tool(SCM)](https://www.softwaretestinghelp.com/version-control-software/) and always deploy the master branch in production. Once deployed, you can regularly [pull](versions#pull) changes and update them to your SCM or revert to them when the need arises. Doing so helps you harness your SCM power for branches, merging conflicting files, reviewing changes, and creating revisions.
+
+Suppose you have a more complex deployment pipeline with one or multiple staging environments with pending changes on each environment. In that case, you can easily track and work on them using development pipelines.
+
+### Setting up Botpress Git Syncing (with GitHub)
+
+1. [Download Botpress](https://botpress.com/download)
+2. Open your terminal and ```cd [PATH]/botpress-v12_20_2-darwin-x64```
+3. Set up GitHub remote tracking
+```
+git init
+git add README.md
+git commit -m "first commit"
+git branch -M main
+git remote add origin git@github.com:[GH_USERNAME]/bp-project.git
+git push -u origin main
+```
+4. You can add a develop and staging branch to fit your workflow
+5. Everytime you make a change in the Conversation Studio or in your preferred code editor, you can commit that change to track changes.
+6. When you are ready to send you chatbot to production, you can ```git merge``` your develop branch in master.
```

Comment:
You mention above that the develop and staging branches are optional, but here you say that we have to merge develop into master in order to deploy in production. I think this could be confusing for those that want to be trunk-based and only use feature/bugfix/hotfix branches. 

What about, instead of saying: `you can ```git merge``` your develop branch in master.`, you change it for something like: `you can ```git merge``` your changes into main.`? 

What do you think?

Reply:
Good call Laurent, thanks for that